src/mmds/src/lib.rs

```diff
@@ -74,39 +74,30 @@ pub fn parse_request(request_bytes: &[u8]) -> Response {
             let response = MMDS
                 .lock()
                 .expect("Failed to build MMDS response due to poisoned lock")
-                .get_value(uri.to_string());
+                .get_value(uri.to_string(), request.headers.accept());
+
             match response {
-                Ok(response) => {
-                    let response_body = response.join("\n");
-                    build_response(
-                        request.http_version(),
-                        StatusCode::OK,
-                        Body::new(response_body),
-                    )
-                }
-                Err(e) => {
-                    match e {
-                        MmdsError::NotFound => {
-                            // NotFound
-                            let error_msg = format!("Resource not found: {}.", uri);
-                            build_response(
-                                request.http_version(),
-                                StatusCode::NotFound,
-                                Body::new(error_msg),
-                            )
-                        }
-                        MmdsError::UnsupportedValueType => {
-                            // InternalServerError
-                            let error_msg =
-                                format!("The resource {} has an invalid format.", uri.to_string());
-                            build_response(
-                                request.http_version(),
-                                StatusCode::InternalServerError,
-                                Body::new(error_msg),
-                            )
-                        }
+                Ok(response_body) => build_response(
+                    request.http_version(),
+                    StatusCode::OK,
+                    Body::new(response_body),
+                ),
+                Err(e) => match e {
+                    MmdsError::NotFound => {
+                        let error_msg = format!("Resource not found: {}.", uri);
+                        build_response(
+                            request.http_version(),
+                            StatusCode::NotFound,
+                            Body::new(error_msg),
+                        )
                     }
-                }
+                    MmdsError::UnsupportedValueType => build_response(
+                        request.http_version(),
+                        StatusCode::NotImplemented,
```

Comment:
Why change this from `InternalServerError` to `NotImplemented`?

Reply:
I thought that it does not make sense. HTTP RFC says: 500 Internal Server Error server error response code indicates that the server encountered an unexpected condition that prevented it from fulfilling the request. `UnsupportedValueType` error is an expected error, which happens every time a guest application tries to retrieve a data store value which can not be flattened (e.g. JSON arrays). This shouldn't happen. The exposed metadata has to be valid for retrieval from guest side, but we can not guarantee what metadata Firecracker customers will expose to guests. Please let me know what you think.

Reply:
Sounds good. It looks like this is a user-facing change. Should we also add it to the changelog?